pkg/services/tree/signature.go

```diff
@@ -132,3 +159,50 @@ func signMessage(m message, key *ecdsa.PrivateKey) error {
 
 	return nil
 }
+
+func roleFromReq(cnr *core.Container, req message) (acl.Role, error) {
+	role := acl.RoleOthers
+	owner := cnr.Value.Owner()
+
+	// No error is expected because `VerifyDataWithSource` checks the signature.
```

Comment:
`VerifyDataWithSource` seems out-of-context here. Why do you mention it? Is it expected to have already been called?

Reply:
in fact, it was just [copied](https://github.com/nspcc-dev/neofs-node/blob/8e8265d5ea4e7c878815b832c523d8d9bc6ecc8c/pkg/services/tree/signature.go#L42-L43), not my code. dropped